classes/CMSCategory.php

```diff
@@ -498,23 +498,38 @@ public static function checkBeforeMove($id_cms_category, $id_parent)
         }
     }
 
-    public static function getLinkRewrite($id_cms_category, $id_lang)
+    /**
+     * Get the rewrite link of the given CMS Category.
+     *
+     * @param int $idCMSCategory CMS Category ID
+     * @param int $idLang Language ID
+     * @param int $idShop Shop ID
+     *
+     * @return bool|mixed
+     */
+    public static function getLinkRewrite($idCMSCategory, $idLang, $idShop = null)
     {
-        if (!Validate::isUnsignedId($id_cms_category) || !Validate::isUnsignedId($id_lang)) {
+        if (!Validate::isUnsignedId($idCMSCategory) || !Validate::isUnsignedId($idLang)) {
             return false;
         }
 
-        if (isset(self::$_links[$id_cms_category . '-' . $id_lang])) {
-            return self::$_links[$id_cms_category . '-' . $id_lang];
+        if (empty($idShop) || !Validate::isUnsignedInt($idShop)) {
+            $idShop = Context::getContext()->shop->id ?
+                Context::getContext()->shop->id :
+                Configuration::get('PS_SHOP_DEFAULT');
+        }
+
+        if (isset(self::$_links[$idCMSCategory . '-' . $idLang . '-' . $idShop])) {
+            return self::$_links[$idCMSCategory . '-' . $idLang . '-' . $idShop];
         }
 
         $result = Db::getInstance()->getRow('
```

Comment:
Why using getRow for only one value ? (even if it was always like that before)

`$linkRewrite = Db::getInstance()->getValue`